### PR TITLE
Migrate to GitHub actions & add MacOS pipeline

### PR DESCRIPTION
**Pull request by aherrmann**
depends on #157

- Adds GitHub actions CI workflows for Linux and MacOS (based on the configuration in rules_haskell)
    All tests worked on MacOS right away apart from the one using busybox - the Nix derivation is not-supported on MacOS.
- Adds CI configuration in `.bazelrc`
- Configures BuildBuddy remote cache and build event protocol on CI
- Removes the buildkite configuration
- Updates the Mergify configuration

### CODE REVIEW

**Review by mboes — APPROVED** [review]

File: .bazelrc [hunk 1]
```diff
@@ -5,3 +5,22 @@ build --crosstool_top=@nixpkgs_config_cc//:toolchain
...
+
+# CI Configuration
+# ----------------
+common:ci --color=no
+build:ci --verbose_failures
+test:ci --test_output=errors
+
+# Use a remote cache during CI
+build:ci --bes_results_url=https://app.buildbuddy.io/invocation/
+build:ci --bes_backend=grpcs://cloud.buildbuddy.io
```
Comment (regnat) [inline .bazelrc:17]: For one second I thought this was remote build. Would have been quite a surprise to see it working

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -1,5 +0,0 @@
-steps:
```
Comment (regnat) [inline .buildkite/pipeline.yml:1]: I think we'll also need to disable buildkite *somewhere* because it currently shows as a failing check in github.

**Review by regnat — APPROVED** [review]
Looks like I'm too late and this just got merged, but :+1: anyways

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (aherrmann) [reply .buildkite/pipeline.yml:1]: Done, I've removed it from the branch protection and archived the pipeline.

**Pull request merged**